### DISCUSSION TITLE
[Semantic API] Should we consider making fieldDescriptors() return a map?

### DISCUSSION THREAD

**Discussion by pubudu91** [ballerina-platform/ballerina-lang#27932] [Ideas]
In the internal types we have for records and objects, fields are kept as a map (`String` key and `BField` value). We previously had a `List` for these as well. But this resulted in people writing util methods all over the place for getting a particular field, given a name. This prompted us to change it to a map. Should we consider doing the same in the record and object type symbols in the Semantic API as well? 

Asking cz I came across similar patterns in lang server as well. e.g., https://github.com/ballerina-platform/ballerina-lang/blob/60c83330c0ebcdbc68d8b6232223053e33d700d9/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/FieldAccessContext.java#L151

**Comment by rasika** [comment]
+1 for the improvement. However we need to be mindful that currently it is an ordered list. Making map should also preserve order of the fields since order is important for some operations such as generating hover markup text.

**Reply by pubudu91** [reply]
Yeah, true. In the internal types we use a `LinkedHashMap` to preserve the order.

**Comment by hasithaa** [comment]
+1

**Comment by pubudu91** [comment]
Great! Will go ahead with the change. Thanks!
